File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FFrameLayout.java
package io.github.gofaith.faithdroid.FViews;

import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

import org.json.JSONArray;
import org.json.JSONTokener;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FFrameLayout extends FView implements AttrGettable, AttrSettable {
    public FrameLayout v;

    public FFrameLayout(UIController controller) {
        parentController = controller;
        v = new FrameLayout(parentController.activity);
        view=v;
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
            // --------------------------------------------------
        }
        return null;
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr){
            // ----------------------------------------------------------------------------
            case "AddView":
                final String childVid = value;
                FView f = parentController.viewmap.get(childVid);
                v.addView(f.view,parseLP(f));
                break;
            case "AddViewAt":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    int pos = Integer.parseInt(array.getString(0));
                    String vid = array.getString(1);
                    FView f1 = parentController.viewmap.get(vid);
                    v.addView(f1.view,pos,parseLP(f1));
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "OnClick":
                v.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Faithdroid.triggerEventHandler(value, "");
                    }
                });
                break;
        }
    }


    public static FrameLayout.LayoutParams parseLP(FView f) {
        FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(f.size[0], f.size[1]);
        lp.gravity=f.layoutGravity;
        lp.leftMargin = f.margin[0];
        lp.topMargin = f.margin[1];
        lp.rightMargin = f.margin[2];
        lp.bottomMargin = f.margin[3];
        return lp;
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FFrameLayout.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.widget.FrameLayout
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.UIController
import org.json.JSONArray
import org.json.JSONTokener

class FFrameLayout : FView, AttrGettable, AttrSettable {
    var v: FrameLayout
    constructor(c:UIController){
        parentController=c
        v=FrameLayout(c.activity)
        view=v
    }

    override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {

        }// --------------------------------------------------
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // ----------------------------------------------------------------------------
            "AddView" -> {
                val f = parentController!!.viewmap[value]
                v.addView(f!!.view, parseLP(f))
            }
            "AddViewAt" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val pos = Integer.parseInt(array.getString(0))
                val vid = array.getString(1)
                val f1 = parentController!!.viewmap[vid]
                v.addView(f1!!.view, pos, parseLP(f1))
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "OnClick" -> v.setOnClickListener { Faithdroid.triggerEventHandler(value, "") }
        }
    }


    companion object {
        fun parseLP(f: FView): FrameLayout.LayoutParams {
            val lp = FrameLayout.LayoutParams(f.size[0], f.size[1])
            lp.gravity = f.layoutGravity
            lp.leftMargin = f.margin[0]
            lp.topMargin = f.margin[1]
            lp.rightMargin = f.margin[2]
            lp.bottomMargin = f.margin[3]
            return lp
        }
    }
}